services/property-svc/internal/property/projections/property_postgres_projection/property_postgres_projection.go

```diff
@@ -0,0 +1,614 @@
+	log := zlog.Ctx(ctx)
+	db := hwdb.GetDB()
+
+	tx, err := db.Begin(ctx)
+	if err != nil {
+		return err, esdb.NackActionRetry
+	}
+
+	defer func() {
+		err := tx.Rollback(ctx)
+		if !errors.Is(err, pgx.ErrTxClosed) {
+			log.Error().Err(err).Msg("rollback failed.")
+		}
+	}()
+
+	propertyRepo := property_repo.New(db).WithTx(tx)
+
+	var payload propertyEventsV1.PropertyCreatedEvent
+	if err := evt.GetJsonData(&payload); err != nil {
+		log.Error().Err(err).Msg("unmarshal failed")
+		return err, esdb.NackActionRetry
+	}
+
+	// Create FieldTypeData
+	fieldTypeDataID, err := propertyRepo.CreateFieldTypeData(ctx)
+	if err := hwdb.Error(ctx, err); err != nil {
+		return err, esdb.NackActionRetry
+	}
+
+	// Create Property
+	propertyID, err := uuid.Parse(payload.ID)
+	if err != nil {
+		return err, esdb.NackActionRetry
+	}
+
+	value, found := pb.SubjectType_value[payload.SubjectType]
+	if !found {
+		return fmt.Errorf("subject_type %s invalid", payload.SubjectType), esdb.NackActionRetry
+	}
+	subjectType := (pb.SubjectType)(value)
+
+	value, found = pb.FieldType_value[payload.FieldType]
+	if !found {
+		return fmt.Errorf("field_type %s invalid", payload.FieldType), esdb.NackActionRetry
+	}
+	fieldType := (pb.FieldType)(value)
+
+	err = propertyRepo.CreateProperty(ctx, property_repo.CreatePropertyParams{
+		ID:              propertyID,
+		SubjectType:     int32(subjectType),
+		FieldType:       int32(fieldType),
+		Name:            payload.Name,
+		Description:     "",
+		IsArchived:      false,
+		FieldTypeDataID: fieldTypeDataID,
+	})
+	if err := hwdb.Error(ctx, err); err != nil {
+		return err, esdb.NackActionRetry
+	}
+
+	if err := tx.Commit(ctx); err != nil {
+		return err, esdb.NackActionRetry
+	}
+
+	return nil, esdb.NackActionUnknown
+}
+
+func (p *Projection) onPropertyDescriptionUpdated(ctx context.Context, evt hwes.Event) (error, esdb.NackAction) {
+	log := zlog.Ctx(ctx)
+
+	var payload propertyEventsV1.PropertyDescriptionUpdatedEvent
+	if err := evt.GetJsonData(&payload); err != nil {
+		log.Error().Err(err).Msg("unmarshal failed")
+		return err, esdb.NackActionRetry
+	}
+
+	err := p.propertyRepo.UpdateProperty(ctx, property_repo.UpdatePropertyParams{
+		ID:          evt.AggregateID,
+		Description: &payload.Description,
+	})
+	err = hwdb.Error(ctx, err)
+	if err != nil {
+		return err, esdb.NackActionRetry
+	}
+
+	return nil, esdb.NackActionUnknown
+}
+
+func (p *Projection) onPropertySetIDUpdated(ctx context.Context, evt hwes.Event) (error, esdb.NackAction) {
+	log := zlog.Ctx(ctx)
+
+	var payload propertyEventsV1.PropertySetIDUpdatedEvent
+	if err := evt.GetJsonData(&payload); err != nil {
+		log.Error().Err(err).Msg("unmarshal failed")
+		return err, esdb.NackActionRetry
+	}
+
+	setID := uuid.NullUUID{UUID: uuid.Nil, Valid: false}
+	if len(payload.SetID) > 0 {
+		parsedID, err := hwutil.ParseNullUUID(&payload.SetID)
+		if err != nil {
+			return err, esdb.NackActionRetry
+		}
+		setID = parsedID
```

Comment:
```suggestion
	if parsedID, err := hwutil.ParseNullUUID(&payload.SetID); err != nil {
		return err, esdb.NackActionRetry
	} else {
		setID = parsedID
	}
```

Reply:
how would this handle the case where a property is removed from a set?

Reply:
If a property is beeing removed then `payload.SetID == ""`. If this is the case we will update to uuid.NullUUID in the db.